common/src/main/java/org/apache/uniffle/common/metrics/MetricReporterFactory.java

```diff
@@ -33,7 +33,7 @@ public static MetricReporter getMetricReporter(RssConf conf, String instanceId)
     }
     Class<?> klass = Class.forName(name);
     Constructor<?> constructor;
-    constructor = klass.getConstructor(conf.getClass(), instanceId.getClass());
+    constructor = klass.getConstructor(RssConf.class, String.class);
```

Comment:
What's the difference from the origin code?

Reply:
In the origin code, if `conf` is an instance of subclass of RssConf, exception will be thrown.  #796